assets/js/components/ModulesList.js

```diff
@@ -0,0 +1,182 @@
+import Link from './Link';
+import ModuleIcon from './ModuleIcon';
+import GenericError from './notifications/generic-error';
+import ModuleSettingsWarning from './notifications/module-settings-warning';
+import VisuallyHidden from './VisuallyHidden';
+import { STORE_NAME as CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ModulesList( { moduleSlugs } ) {
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const modulesData = useSelect( ( select ) => select( CORE_MODULES ).getModules() ?? {} );
+
+	/**
+	 * Handle setup module click event.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param {string} slug Module slug.
+	 */
+	const handleSetupModule = useCallback( async ( slug ) => {
+		try {
+			const { response } = await activateModule( slug );
+			await trackEvent(
+				`${ slug }_setup`,
+				'module_activate',
+				slug,
+			);
+
+			// Redirect to ReAuthentication URL
+			global.location.assign( response.moduleReauthURL );
+		} catch ( err ) {
+			showErrorNotification( GenericError, {
+				id: 'setup-module-error',
+				title: __( 'Internal Server Error', 'google-site-kit' ),
+				description: err.message,
+				format: 'small',
+				type: 'win-error',
+			} );
+		}
+	}, [ activateModule ] );
+
+	// Filter specific modules
+	const moduleObjects = Array.isArray( moduleSlugs ) && moduleSlugs.length
+		? moduleSlugs
+			.filter( ( slug ) => modulesData.hasOwnProperty( slug ) )
+			.reduce( ( acc, slug ) => ( { ...acc, [ slug ]: modulesData[ slug ] } ), {} )
+		: modulesData;
+
+	// Filter out internal modules.
+	const modules = Object.values( moduleObjects ).filter( ( module ) => ! module.internal );
+
+	// Map of slug => name for every module that is active and completely set up.
+	const completedModuleNames = modules
+		.filter( ( module ) => module.active && module.connected )
+		.reduce( ( completed, module ) => {
+			completed[ module.slug ] = module.name;
+			return completed;
+		}, {} );
+
+	// Sort modules and exclude those that required other modules to be set up.
+	// Logic still in place below in case we want to add blocked modules back.
+	const sortedModules = modules
+		.filter( ( module ) => 0 === module.dependencies.length )
+		.sort( ( module1, module2 ) => module1.sort - module2.sort );
+
+	return (
+		<div className="googlesitekit-modules-list">
+			{ map( sortedModules, ( module ) => {
+				let blockedByParentModule = false;
+				let parentBlockerName = '';
+				const {
+					slug,
+					name,
+					connected,
+					active,
+					dependencies,
+				} = module;
+				const setupComplete = connected && active;
+
+				// Check if required modules are active.
+				if ( 0 < dependencies.length ) {
+					dependencies.forEach( ( requiredModule ) => {
+						if ( completedModuleNames[ requiredModule ] ) {
+							blockedByParentModule = true;
+							parentBlockerName = completedModuleNames[ requiredModule ];
+						}
+					} );
+				}
+
```

Comment:
I removed the filter but I kept the `blockedByParentModule `. Let me explain why:
As I understand, the `ModulesList` component will show a list of modules that can be activated. The modules which can't be activated because their dependency is not activated, should show a greyed out message asking to activate the dependency first. 

I made a few changes to how `blockedByParentModule` was set and the surrounding code. 

I'd like to request you to pull the latest changes, start storybook and go to the `ModulesList` story. Once you go you will see all the modules including `TagManager`  and `Optimize` since `provideModules` activates `Analytics` by default.
Now go to `fixtures.json` and set `active` for `Analytics` to false, wait for storybook to reload, and you will see that `TagManager` and `Optimize` can't be activated and shows the desired message. 

Do let me know if this is the desired requirement for this Component. If not, then I will remove `blockedByParentModule ` as you suggested above. In that case I have two questions:
1. Will the modules with *unactivated* dependencies be shown by default?
2. Will the aforementioned modules show the greyed out message asking to activate `dependency`? Or will they be allowed to be activated unconditionally?
3. Will the aforementioned modules be not shown?

Reply:
Regarding my comment above, I wasn't suggesting we remove the filter, I was only suggesting we update that part (as it's relevant to the ACs) to be a bit more clear either by using separate variables or inline it perhaps to not have a variable called "sorted" which also filters. 

Thanks for the additional context. It's important to understand that the issue was defined specifically with the removal of `blockedByParentModule` behavior as mentioned in the AC because it is not possible to work otherwise with the current/legacy version of the component. To do otherwise means that we would be introducing a behavioral change to the component where it is not called for and not currently present. 

While this may have been a bug and I can appreciate your desire to fix it, a conflict with the AC should be raised with @felixarntz as soon as possible. As it is, I think the issue is clear about what should be done but we can certainly wait for @felixarntz to give the final word.

Reply:
@kostyalmm @aaemnnosttv Indeed the ability of showing modules with inactive dependencies greyed out is legacy behavior, which used to be the case, but was intentionally changed a while ago. Unfortunately the code back then wasn't fully removed which is probably what created the confusion here. The way things currently look after your latest commits should be good.